File: CS/MvpvmNavigation/Views/ModuleB/View2EditForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using MvpvmNavigation.ViewModels;

namespace MvpvmNavigation.Views {
    public partial class View2EditForm : BaseModuleControl {
        public View2EditForm()
            : base(CreateViewModel<ModuleBEditFormViewModel>) {
            InitializeComponent();
            BindCommands();
        }

        void BindCommands() {
            buttonOK.BindCommand(() => ViewModel.OK(), ViewModel);
            buttonCancel.BindCommand(() => ViewModel.Cancel(), ViewModel);
        }

        public ModuleBEditFormViewModel ViewModel {
            get { return GetViewModel<ModuleBEditFormViewModel>(); }
        }
    }
}

File: VB/MvpvmNavigation/Views/ModuleB/View2EditForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports MvpvmNavigation.ViewModels

Namespace MvpvmNavigation.Views
    Partial Public Class View2EditForm
        Inherits BaseModuleControl

        Public Sub New()
            MyBase.New(AddressOf CreateViewModel(Of ModuleBEditFormViewModel))
            InitializeComponent()
            BindCommands()
        End Sub

        Private Sub BindCommands()
            buttonOK.BindCommand(Sub() ViewModel.OK(), ViewModel)
            buttonCancel.BindCommand(Sub() ViewModel.Cancel(), ViewModel)
        End Sub

        Public ReadOnly Overloads Property ViewModel() As ModuleBEditFormViewModel
            Get
                Return GetViewModel(Of ModuleBEditFormViewModel)()
            End Get
        End Property
    End Class
End Namespace
